[2025-01-02T02:02:48.933+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-02T02:02:48.990+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: daily_news_dag.daily_news_etl_run scheduled__2025-01-02T01:50:00+00:00 [queued]>
[2025-01-02T02:02:49.022+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: daily_news_dag.daily_news_etl_run scheduled__2025-01-02T01:50:00+00:00 [queued]>
[2025-01-02T02:02:49.023+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 1
[2025-01-02T02:02:49.049+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): daily_news_etl_run> on 2025-01-02 01:50:00+00:00
[2025-01-02T02:02:49.076+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=75) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-01-02T02:02:49.078+0000] {standard_task_runner.py:72} INFO - Started process 84 to run task
[2025-01-02T02:02:49.093+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'daily_news_dag', 'daily_news_etl_run', 'scheduled__2025-01-02T01:50:00+00:00', '--job-id', '56', '--raw', '--subdir', 'DAGS_FOLDER/daily_news_dag.py', '--cfg-path', '/tmp/tmp3osgzox1']
[2025-01-02T02:02:49.098+0000] {standard_task_runner.py:105} INFO - Job 56: Subtask daily_news_etl_run
[2025-01-02T02:02:49.369+0000] {task_command.py:467} INFO - Running <TaskInstance: daily_news_dag.daily_news_etl_run scheduled__2025-01-02T01:50:00+00:00 [running]> on host ade868307fae
[2025-01-02T02:02:51.395+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='daily_news_dag' AIRFLOW_CTX_TASK_ID='daily_news_etl_run' AIRFLOW_CTX_EXECUTION_DATE='2025-01-02T01:50:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-02T01:50:00+00:00'
[2025-01-02T02:02:51.397+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-02T02:02:51.687+0000] {daily_news_etl.py:130} INFO - Starting ETL process
[2025-01-02T02:02:51.688+0000] {daily_news_etl.py:30} INFO - Fetching news for business category
[2025-01-02T02:02:54.225+0000] {daily_news_etl.py:45} INFO - Creating a dataframe for business category
[2025-01-02T02:02:54.266+0000] {daily_news_etl.py:60} INFO - Analysing and adding sentiment to the dataframe...
[2025-01-02T02:02:54.268+0000] {log.py:232} WARNING - No model was supplied, defaulted to distilbert/distilbert-base-uncased-finetuned-sst-2-english and revision 714eb0f (https://huggingface.co/distilbert/distilbert-base-uncased-finetuned-sst-2-english).
Using a pipeline without specifying a model name and revision in production is not recommended.
[2025-01-02T02:02:54.847+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scripts/daily_news_etl.py", line 135, in main
    df = etl.analyse_sentiment(df)
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scripts/daily_news_etl.py", line 61, in analyse_sentiment
    classification = pipeline('sentiment-analysis', truncation=True)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/pipelines/__init__.py", line 940, in pipeline
    framework, model = infer_framework_load_model(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/pipelines/base.py", line 240, in infer_framework_load_model
    raise RuntimeError(
RuntimeError: At least one of TensorFlow 2.0 or PyTorch should be installed. To install TensorFlow 2.0, read the instructions at https://www.tensorflow.org/install/ To install PyTorch, read the instructions at https://pytorch.org/.
[2025-01-02T02:02:54.876+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=daily_news_dag, task_id=daily_news_etl_run, run_id=scheduled__2025-01-02T01:50:00+00:00, execution_date=20250102T015000, start_date=20250102T020248, end_date=20250102T020254
[2025-01-02T02:02:54.920+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-02T02:02:54.921+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 56 for task daily_news_etl_run (At least one of TensorFlow 2.0 or PyTorch should be installed. To install TensorFlow 2.0, read the instructions at https://www.tensorflow.org/install/ To install PyTorch, read the instructions at https://pytorch.org/.; 84)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scripts/daily_news_etl.py", line 135, in main
    df = etl.analyse_sentiment(df)
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/scripts/daily_news_etl.py", line 61, in analyse_sentiment
    classification = pipeline('sentiment-analysis', truncation=True)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/pipelines/__init__.py", line 940, in pipeline
    framework, model = infer_framework_load_model(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/transformers/pipelines/base.py", line 240, in infer_framework_load_model
    raise RuntimeError(
RuntimeError: At least one of TensorFlow 2.0 or PyTorch should be installed. To install TensorFlow 2.0, read the instructions at https://www.tensorflow.org/install/ To install PyTorch, read the instructions at https://pytorch.org/.
[2025-01-02T02:02:55.007+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-02T02:02:55.056+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-02T02:02:55.069+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
